Assets/HoloToolkit/Build/Editor/BuildDeployWindow.cs

```diff
@@ -0,0 +1,581 @@
+        }
+
+        private bool UninstallApp(string appName, string targetDevice)
+        {
+            // Connection info
+            var connectInfo = new BuildDeployPortal.ConnectInfo(targetDevice, deviceUser, devicePassword);
+
+            // Kick off the install
+            Debug.Log("Uninstall build: " + targetDevice);
+            return BuildDeployPortal.UninstallApp(appName, connectInfo);
+        }
+
+        private void UpdateBuilds()
+        {
+            this.builds.Clear();
+
+            try
+            {
+                List<string> appPackageDirectories = new List<string>();
+                string[] buildList = Directory.GetDirectories(BuildDirectory_FullPath);
+                foreach (string appBuild in buildList)
+                {
+                    string appPackageDirectory = appBuild + @"\AppPackages";
+                    if (Directory.Exists(appPackageDirectory))
+                    {
+                        appPackageDirectories.AddRange(Directory.GetDirectories(appPackageDirectory));
+                    }
+                }
+                IEnumerable<string> selectedDirectories = from string directory in appPackageDirectories orderby Directory.GetLastWriteTime(directory) descending select Path.GetFullPath(directory);
+                this.builds.AddRange(selectedDirectories);
+            }
+            catch (DirectoryNotFoundException)
+            {
+            }
+
+            timeLastUpdatedBuilds = Time.realtimeSinceStartup;
+        }
+
+        void Update()
+        {
+            if ((Time.realtimeSinceStartup - timeLastUpdatedBuilds) > kUpdateBuildsPeriod)
+            {
+                UpdateBuilds();
+            }
+        }
+
+        private static string GetEditorPref(string key, string defaultValue)
+        {
+            if (EditorPrefs.HasKey(key))
+            {
+                return EditorPrefs.GetString(key);
+            }
+            else
+            {
+                EditorPrefs.SetString(key, defaultValue);
+                return defaultValue;
+            }
+        }
+
+        private static bool GetEditorPref(string key, bool defaultValue)
+        {
+            if (EditorPrefs.HasKey(key))
+            {
+                return EditorPrefs.GetBool(key);
+            }
+            else
+            {
+                EditorPrefs.SetBool(key, defaultValue);
+                return defaultValue;
+            }
+        }
+
+        public static string[] ParseIPList(string IPs)
+        {
+            string[] IPlist = { };
+
+            if (IPs == null || IPs == "")
+                return IPlist;
+
+            string[] separators = { ";", " " };
+            IPlist = IPs.Split(separators, System.StringSplitOptions.RemoveEmptyEntries);
+            return IPlist;
+        }
+
+        static string FinalizeIP(string ip)
+        {
+            // If it's local, add the port
+            if (ip == "127.0.0.1")
+            {
+                ip += ":10080";
```

Comment:
Nit: In a future rev., we should look this up in the registry as this can be configured differently.